openmrs/omod/src/main/java/org/projectbuendia/openmrs/web/controller/ProfileManager.java

```diff
@@ -48,34 +42,70 @@
 /** The controller for the profile management page. */
 @Controller
 public class ProfileManager {
-    static Log log = LogFactory.getLog(ProfileManager.class);
-    final File PROFILE_DIR = new File("/usr/share/buendia/profiles");
-    final String VALIDATE_CMD = "buendia-profile-validate";
-    final String APPLY_CMD = "buendia-profile-apply";
+    protected static Log log = LogFactory.getLog(ProfileManager.class);
+    private final String PROFILE_DIR_PATH = "/usr/share/buendia/profiles";
+    private final String VALIDATE_CMD = "buendia-profile-validate";
+    private final String APPLY_CMD = "buendia-profile-apply";
+    private File profileDir;
+
+    @ModelAttribute
+    public void getInitializedMyObject() {
+        Properties config = Context.getRuntimeProperties();
+        String configProfileDir = config.getProperty("profile_manager.profile_dir");
+        if(configProfileDir != null){
+            profileDir = new File(configProfileDir);
+        } else {
+            profileDir = new File(PROFILE_DIR_PATH);
+        }
+    }
 
     @RequestMapping(value = "/module/projectbuendia/openmrs/profiles", method = RequestMethod.GET)
     public void get(HttpServletRequest request, ModelMap model) {
-        List<FileInfo> files = new ArrayList<>();
-        for (File file : PROFILE_DIR.listFiles()) {
-            files.add(new FileInfo(file));
+        String currentProfile = Context.getAdministrationService().getGlobalProperty(GlobalProperties.CURRENT_PROFILE);
+        model.addAttribute("currentProfile", currentProfile);
+        model.addAttribute("authorized", authorized());
+        model = queryParamsToModelAttr(request, model);
+        if(!profileDir.exists()){
+            model.addAttribute("success", false);
+            model.addAttribute("message", "The Profile Manager directory does not exist. It must reside on " +
+                    "/usr/share/buendia/profiles or on a directory specified in openmrs-runtime.properties " +
+                    "file under profile_manager.profile_dir directive.");
+        } else {
+            model.addAttribute("profiles", listProfiles());
         }
-        Collections.sort(files, new Comparator<FileInfo>() {
-            public int compare(FileInfo a, FileInfo b) {
-                return -a.modified.compareTo(b.modified);
-            }
-        });
+    }
 
-        model.addAttribute("user", Context.getAuthenticatedUser());
-        model.addAttribute("profiles", files);
-        model.addAttribute("currentProfile",
-            Context.getAdministrationService().getGlobalProperty(
-                GlobalProperties.CURRENT_PROFILE));
-        model.addAttribute("authorized", authorized());
+    private ModelMap queryParamsToModelAttr(HttpServletRequest request, ModelMap model){
+        Map params = request.getParameterMap();
```

Comment:
A for loop does the job nicely here, and using the parameterized type `Map<String, String[]>` gives better type safety:
```
        for (Map.Entry<String, String[]> entry : params.entrySet()) {
            model.addAttribute(entry.getKey(), entry.getValue()[0]);
        }
```